src/main/java/org/ossgang/commons/mapbackeds/Mapbackeds.java

```diff
@@ -0,0 +1,223 @@
+     * </pre>
+     *
+     * NOTE: Currently, there is no check for null values. Therefore, only partially filled objects (or even empty
+     * ones) would be allowed. Fields which are not set, would return {@code null} in this case. However, be aware that
+     * methods returning primitive types would throw in this case! This behavior might change in future versions.
+     *
+     * @param backedInterface the interface which shall be backed by a map of values.
+     * @throws IllegalArgumentException if the given class is not an interface
+     */
+    public static <M> Builder<M> builder(Class<M> backedInterface) {
+        return new Builder<>(backedInterface);
+    }
+
+    /**
+     * Creates a mapbacked object with the given interface, backed by the given map of field Values. No Consistency
+     * checks are done! This behavior might change in future versions.
+     */
+    public static <M> M from(Class<M> backedInterface, Map<String, Object> fieldValues) {
+        /*
+         * TODO: add consistency checks!
+         */
+        return proxy(backedInterface, new MapbackedObject(backedInterface, fieldValues));
+    }
+
+    /**
+     * Retrieves the internal map of the given mapbacked object. As this will throw, if the given object is not
+     * mapbacked, it is recommended to check before if the object is a mapbacked, by using the
+     * {@link #isMapbacked(Object)} method.
+     *
+     * @param the mapbacked object from which to retrieve the internal map
+     * @return the internal map of the object
+     * @throws IllegalArgumentException if the given object is not a mapbacked object
+     */
+    public static Map<String, Object> mapOf(Object object) {
+        Optional<MapbackedObject> handler = handlerFrom(object);
+        if (handler.isPresent()) {
+            return handler.get().fieldValues();
+        }
+        throw new IllegalArgumentException("The given object seems not to be a Mapbacked object: " + object);
+    }
+
+    /**
+     * Checks, if the given object is mapbacked
+     *
+     * @param the object to check
+     * @return {@code true} if the given object is mapbacked, {@code false} otherwise
+     */
+    public static boolean isMapbacked(Object object) {
+        return handlerFrom(object).isPresent();
+    }
+
+    private static Optional<MapbackedObject> handlerFrom(Object object) {
+        if (!(object instanceof Proxy)) {
+            return Optional.empty();
+        }
+
+        InvocationHandler handler = Proxy.getInvocationHandler(object);
+        if (!(handler instanceof MapbackedObject)) {
+            return Optional.empty();
+        }
+
+        return Optional.of((MapbackedObject) handler);
+    }
+
+    public static class Builder<M> {
+
+        private final Class<M> backedInterface;
+        private final Set<Method> fieldMethods;
+        private final Map<String, Object> mapBuilder = new HashMap<>();
+
+        /*
+         * TODO: Introduce consistency checks
```

Comment:
Are these TODOs still to do? at least some checks seem to be here?

Reply:
It is anyhow not nice to have those todos in the coed. Therefore, I created a new issue with the ideas #47  and removed the todos from the code.